### PR TITLE
Begin to close the gap between this and scheduled-function

### PR DESCRIPTION
**Pull request by zachberger**

### CODE REVIEW

**Comment by zachberger** [comment]
After these changes, scheduled-function can be implemented using event-function

File: main.tf [hunk 1]
```diff
@@ -47,6 +48,10 @@ resource "google_cloudfunctions_function" "main" {
   event_trigger {
     event_type = "${var.event_trigger["event_type"]}"
     resource   = "${var.event_trigger["resource"]}"
+
+    failure_policy {
+      retry = "${var.event_trigger_failure_policy_retry}"
```
Comment (zachberger) [inline main.tf]: I tried doing something similar to `${var.event_trigger["resource"]["failure_policy"]["retry"]}` but it failed. Is there a preferred way to reference nested maps?

File: variables.tf [hunk 2]
```diff
@@ -79,3 +79,33 @@ variable "timeout_s" {
...
+
+variable "bucket_name" {
+  type        = "string"
+  default     = ""
+  description = "The name to apply to the bucket. Will default to a string of the function name."
+}
+
+variable "bucket_force_destroy" {
+  type        = "string"
+  default     = "false"
```
[the remark this answers is not in the document]
Comment (zachberger) [reply variables.tf]: I'm ok with that, I set it to false to maintain existing behavior.

File: main.tf [hunk 3]
```diff
@@ -21,11 +21,12 @@ data "archive_file" "main" {
 }
 
 resource "google_storage_bucket" "main" {
-  name          = "${var.name}"
+  name          = "${coalesce(var.bucket_name, var.name)}"
```
[the remark this answers is not in the document]
Comment (zachberger) [reply main.tf]: The existing scheduled function lets you specify them separately. I'm fine with it either way, was just trying to maintain existing functionality.

File: main.tf (hunk 1 above)
Comment (zachberger) [reply main.tf]: I'll give this a try.

File: variables.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (aaron-lane) [reply variables.tf]: Oh, that's fair. Let's leave it as `false`, then.

File: main.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (aaron-lane) [reply main.tf]: Okay, that makes sense.

File: main.tf (hunk 1 above)
Comment (zachberger) [reply main.tf]: Coming back to this topic, I'm not sure how an array of maps would work or help here. Could you elaborate?

**Comment by zachberger** [comment]
@aaron-lane what do you think about removing the default value for the cloud function's runtime? It seems like we'll need to update it frequently with the seemingly constant deprecations. I'm also not convinced `nodejs` is the most commonly used runtime.

File: main.tf (hunk 1 above)
Comment (aaron-lane) [reply main.tf]: `event_trigger` is a "block", which is actually equivalent to a list of maps. However, the provider [limits the list to 1 element](https://github.com/terraform-providers/terraform-provider-google/blob/v2.9.1/google/resource_cloudfunctions_function.go#L231), which means we should be able to pass it like the following without specifically fetching each key-value pair:

```hcl
resource "google_cloudfunctions_function" "main" {
  # ...
  event_trigger = ["${var.event_trigger}"]
  # ...
}
```

**Comment by aaron-lane** [comment]
I'm okay with dropping the default; I just went with it since it was the default language in many CloudFunctions examples.

File: main.tf (hunk 1 above)
Comment (zachberger) [reply main.tf]: I suppose I'm conflicted here. With us defining each as its own variable the user is able to look at the module's variables and know exactly how to configure it, were as if we leave `event_trigger` as a map the user then needs to lookup the format of this specific map.

**Comment by zachberger** [comment]
What do you think about leaving refactoring `event_trigger` for a future time?

File: main.tf (hunk 1 above)
Comment (aaron-lane) [reply main.tf]: Fair. Alternatively, let's break up the `event_trigger` map variable to individual variables.

**Comment by aaron-lane** [comment]
@zachberger sure, we can do that.

**Pull request merged**